### PR TITLE
Add nhk-program-search extension

### PR DESCRIPTION
**Pull request by sitogi**
## Description

This is an extension for displaying and searching the program schedule of NHK, Japan's public broadcaster.

### List Programs

nhk-program-search-1

### Program Detail

nhk-program-search-2

### Search by Genre

nhk-program-search-3

### Search by Keyword

nhk-program-search-4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/nhk-program-search/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# nhk-program-search Changelog
```
Comment (pernielsentikaer) [inline extensions/nhk-program-search/CHANGELOG.md:1]:
```suggestion
# NHK Program Search Changelog
```

File: extensions/nhk-program-search/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# nhk-program-search
```
Comment (pernielsentikaer) [inline extensions/nhk-program-search/README.md:1]:
```suggestion
# NHK Program Search
```

File: extensions/nhk-program-search/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+NHK Program Search
+
+## Getting Started
+
+This Extension uses the API provided by NHK. API Key is required to use this Extension.  
+https://api-portal.nhk.or.jp/service-guide
+
+## Credit
+
+情報提供:ＮＨＫ
```
Comment (pernielsentikaer) [inline extensions/nhk-program-search/README.md:12]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension. If the locale might affect functionality (e.g. using the correct unit of measurement), please use the preferences API.

File: extensions/nhk-program-search/package.json [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+          "title": "宮崎",
+          "value": "450"
+        },
+        {
+          "title": "鹿児島",
+          "value": "460"
+        },
+        {
+          "title": "沖縄",
+          "value": "470"
```
Comment (pernielsentikaer) [inline extensions/nhk-program-search/package.json:291]: Can we make title here english too

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/nhk-program-search/README.md (hunk 3 above)
Comment (sitogi) [reply extensions/nhk-program-search/README.md]: https://api-portal.nhk.or.jp/service-guide

At the bottom of the terms of use for this API, there is a note that says "Display the string '情報提供:ＮＨＫ'", so would it be okay to also keep the Japanese string just to be safe?

image

The English translation of the above image.

> "About Displaying Credits"
> Please ensure to display the following credit on sites and applications utilizing this service.
> When used as standalone data: 「情報提供:ＮＨＫ」

**Comment by sitogi** [comment]
@pernielsentikaer 
Thank you for your review!
We have addressed your feedback. Please check again!

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Can we do the same for items in the dropdown?
Raycast is not ready for localization

nhk-program-search 2024-05-21 at 09 29 31

**Comment by sitogi** [comment]
@pernielsentikaer 

> Can we do the same for items in the dropdown?

OK. I will handle it.

I will also change the genre selection to English.

image

However, please note that this API retrieves the program listings from the Japanese public broadcaster, and it does not support English. As a result, all search results will be in Japanese.

Is this acceptable?

**Comment by sitogi** [comment]
@pernielsentikaer
I have made corrections based on the feedback received. Please review them.

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**